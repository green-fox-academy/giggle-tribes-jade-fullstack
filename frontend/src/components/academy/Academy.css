div.academy {
  grid-area: 3 / 1 / span 1 / span 2;
  display: grid;
  grid-template-columns: 140px 1fr;
  grid-template-rows: 140px 1fr;
  background-color: rgba(201, 204, 205, 0.8);
}
img.academy {
  margin: 20px;
  height: 100px;
  width: 100px;
  grid-area: 1 / 1 /1 /1;
}
div.description {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 20px;
  grid-area: 1 / 2;
}
div.academyinfo {
  display: flex;
  flex-direction: column;
  justify-content: start;
  align-items: center;
  margin: 5px;
  grid-area: 2 / 1;
}
img.inlineIcon {
  height: 15px;
  max-width: 15px;
}
