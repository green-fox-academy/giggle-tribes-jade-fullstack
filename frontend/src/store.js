import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';

import { userReducer } from './reducers/userReducer';
import { kingdomReducer } from './reducers/kingdomReducer';
import { buildingsReducer } from './reducers/buildingsReducer';
import { resourcesReducer } from './reducers/resourcesReducer';
import { troopsReducer } from './reducers/troopsReducer';
import { tokenReducer } from './reducers/tokenReducer';
import { errorReducer } from './reducers/errorReducer';

function root(state = {}, action) {
  return {
    user: userReducer(state.user, action),
    kingdom: kingdomReducer(state.kingdom, action),
    buildings: buildingsReducer(state.buildings, action),
    resources: resourcesReducer(state.resources, action),
    troops: troopsReducer(state.troops, action),
    token: tokenReducer(state.token, action),
    error: errorReducer(state.error, action),
  };
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const store = createStore(
  root,
  /* preloadedState, */ composeEnhancers(applyMiddleware(thunk))
);

export { store };
