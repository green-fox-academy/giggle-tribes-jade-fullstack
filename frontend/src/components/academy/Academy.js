import React, { useState, useEffect, createRef } from 'react';
import PropTypes from 'prop-types';

import './Academy.css';
import Barrack from '../../assets/buildings/barracks.svg';
import Attack from '../../assets/troops/attack1.png';
import Defence from '../../assets/troops/defence1.png';
import Food from '../../assets/sources/FoodIcon.svg';
import Button from './CreateTroopButton';

export default function Academy({ kingdomID }) {
  return (
    <div className="academy">
      <img className="academy" src={Barrack} alt="academy icon" />
      <div className="description">
        <p>
          You can create troops in your Academy. The higher level your Academy
          is, the stronger your troops are.
        </p>
        <p>
          Every level increases 1{' '}
          <img className="inlineIcon" src={Attack} alt="attack icon" /> and 1{' '}
          <img className="inlineIcon" src={Defence} alt="defence icon" /> of the
          Troops.
        </p>
        <p>
          Every Troop eats 1{' '}
          <img className="inlineIcon" src={Food} alt="food icon" /> every
          minute.
        </p>
      </div>
      <div className="academyinfo">
        <p>Academy</p>
        <p>{'Level 1'}</p>
      </div>
      <Button />
    </div>
  );
}

Academy.propTypes = {
  kingdomID: PropTypes.any.isRequired,
};
