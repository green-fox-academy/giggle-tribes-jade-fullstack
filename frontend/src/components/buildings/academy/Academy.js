import React from 'react';

import './Academy.css';
import Barrack from '../../../assets/buildings/barracks.svg';
import Attack from '../../../assets/troops/attack1.png';
import Defence from '../../../assets/troops/defence1.png';
import Food from '../../../assets/sources/FoodIcon.svg';
import Button from './CreateTroopButton';
import Error from '../../Error';

function Academy() {
  const academyLevel = 1;
  return (
    <div className="academy">
      <img className="academy" src={Barrack} alt="academy icon" />
      <div className="description">
        <p>
          You can create troops in your Academy. The higher level your Academy
          is, the stronger your troops are.
        </p>
        <p>
          Every level increases 1{' '}
          <img className="inlineIcon" src={Attack} alt="attack icon" /> and 1{' '}
          <img className="inlineIcon" src={Defence} alt="defence icon" /> of the
          Troops.
        </p>
        <p>
          Every Troop eats 1{' '}
          <img className="inlineIcon" src={Food} alt="food icon" /> every
          minute.
        </p>
      </div>
      <div className="academyinfo">
        <p>Academy</p>
        <p>{'Level ' + academyLevel}</p>
      </div>
      <Button />
      <Error />
    </div>
  );
}

export default Academy;
