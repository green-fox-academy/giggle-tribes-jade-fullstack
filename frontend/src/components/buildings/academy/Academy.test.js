import React from 'react';
import { Provider } from 'react-redux';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';

import Academy from './Academy';
import { store } from '../../../store';

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('matches snapeshot', async () => {
  await act(async () => {
    render(
      <Provider store={store}>
        <Academy />
      </Provider>,
      container
    );
  });

  expect(container.innerHTML).toMatchSnapshot();
});
