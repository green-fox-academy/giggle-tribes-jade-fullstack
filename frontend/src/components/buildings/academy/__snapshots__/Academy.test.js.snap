// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapeshot 1`] = `"<div class=\\"academy\\"><img class=\\"academy\\" src=\\"barracks.svg\\" alt=\\"academy icon\\"><div class=\\"description\\"><p>You can create troops in your Academy. The higher level your Academy is, the stronger your troops are.</p><p>Every level increases 1 <img class=\\"inlineIcon\\" src=\\"attack1.png\\" alt=\\"attack icon\\"> and 1 <img class=\\"inlineIcon\\" src=\\"defence1.png\\" alt=\\"defence icon\\"> of the Troops.</p><p>Every Troop eats 1 <img class=\\"inlineIcon\\" src=\\"FoodIcon.svg\\" alt=\\"food icon\\"> every minute.</p></div><div class=\\"academyinfo\\"><p>Academy</p><p>Level undefined</p></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text addtroopBTN\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\"><img class=\\"addtroop\\" src=\\"addTroop-removebg-preview.png\\" alt=\\"troop icon\\"><div><p>create troop level 1</p><p>10 <img class=\\"inlineIcon\\" src=\\"GoldIcon.svg\\" alt=\\"gold icon\\"> 1:00</p></div></span><span class=\\"MuiTouchRipple-root\\"></span></button><p></p></div>"`;
